lib/models/notification.ts

```diff
@@ -49,41 +49,41 @@ export const NotificationBase = t.interface({
 
 // Email Notification
 
-export const NotificationChannelEmail = t.intersection([
-  t.interface({
-    addressSource: NotificationAddressSource,
-    toAddress: EmailAddress
-  }),
-  t.partial({
-    fromAddress: EmailAddress
-  })
-]);
+export const NotificationChannelEmail = t.interface({
+  [NotificationChannelEnum.EMAIL]: t.intersection([
```

Comment:
I don't understand, this type describes a `NotificationChannelEmail`, why it's an object with a single constant key? (`NotificationChannelEnum.EMAIL`) - why `NotificationChannelEmail` is not `t.intersection([...])` ? (same below for `NotificationChannelWebhook`)